### DISCUSSION TITLE
Welcome to gams-ide Discussions!

### DISCUSSION THREAD

**Discussion by chrispahm** [chrispahm/gams-ide#31] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by SLUTorbjornJansson** [comment]
Hi Christoph. Torbjörn Jansson, from SLU here. I started using the gams-ide now, and wonder if you can change the gams working directory? In CAPRI, we have the structure git-root/gams/main-file.gms, and when I open (in VS Code) the git-root folder, the relative include file paths don't work by default. I need to open the folder gams in VS Code instead. But then I lose the GIT integration. I see that your program sends working directory to gams, but it sends the directory I opened in VS Code. I would like to modify that...

Another issue is the possibility to send command line arguments to the execution (play) button. Can I already do that?

Cheers, and thanks for a very helpful extension.

**Comment by chrispahm** [comment]
Hi Torbjörn,
Thanks for trying out the extension! Yes, everything is configurable, but the discoverability needs to improve 😊 I've just made a new release that should simplify onboarding for CAPRI, so it should apply the most relevant settings automatically.

1. Install the new release → https://github.com/chrispahm/gams-ide/releases/tag/v0.0.21
2. Open a VSCode Workspace in the git root folder for CAPRI, the one where the folder `gams`, `GUI`, etc. are
3. Open the project setup wizard by pressing `Ctrl-Shift-P`, then typing `GAMS: Run Project Setup Wizard`
<img width="644" height="71" alt="image" src="https://github.com/user-attachments/assets/754bfb41-0976-49a6-9b75-eaf9bafcc73d" />

4. This should open the following screen in the sidebar (sidebar is left by default, you can drag it into the right sidebar which is what I prefer)
<img width="420" height="551" alt="image" src="https://github.com/user-attachments/assets/8af63769-ec86-4da0-a450-b01e4e971bfa" />

5. Click on **Use Default CAPRI Settings** → this should take the `gams exe`, `--task`, `workDir`, `scrDir` etc. settings from the `default.ini`. My version has compilation errors, which is expected though. The only thing relevant is that this throws the same error as the GUI when clicking compile, or no error if the GUI also has no errors

<img width="361" height="693" alt="image" src="https://github.com/user-attachments/assets/e80170a9-43da-4443-a898-d0a00d830296" />

6. Once you click on **Done**, there will be a `.vscode/settings.json` file at the root of your workspace, which stores all relevant settings. This is where you can also manually change the workdir, curdir etc.
<img width="1117" height="689" alt="image" src="https://github.com/user-attachments/assets/fd9f49cb-edf6-4fd7-baf4-88543b42a89b" />

Also, the sidebar will have a **Settings** section that shows a preview of the current GAMS command. You can manually change it if you uncheck `Auto-sync command with GUI`. Note that if you change the `settings.json` manually, without unchecking `Auto-sync command with GUI`, it will always override your changes.

<img width="308" height="835" alt="image" src="https://github.com/user-attachments/assets/0c7fb104-8c06-4b2d-b96e-e475871b7326" />

**Note: If you change something in the GUI, you need to save (any) .gms file in VSCode in order for the command preview to refresh, since we're not live tracking changes to the .ini**

Let me know if there are any bugs or issues, so we can resolve them asap!
Best
Christoph
